TaboolaSDKExampleSwift/TaboolaSDKExampleSwift/SDK Native/NativeViewController.swift

```diff
@@ -10,60 +10,110 @@ import Foundation
-                self?.taboolaItem = item
-                self?.updateTaboolaUI()
-            }
+        taboolaUnit?.fetchContent(onSuccess: {[weak self] (response) in
+            self?.itemsArray = response?.items
+            self?.collectionView.reloadData()
         }) { (error) in
             print(error.debugDescription)
         }
     }
     
-    func updateTaboolaUI() {
-        if let item = taboolaItem {
-            item.initThumbnailView(imageView)
-            item.initDescriptionView(descriptionLabel)
-            item.initBrandingView(brandingLabel)
-            item.initTitleView(titleLabel)
-        }
+    func fetchNextPage() {
+        taboolaUnit?.fetchContent(onSuccess: { (response) in
+            if let response = response {
+                guard let newItems = response.items else { return }
+                self.collectionView.performBatchUpdates({
+                    guard let resultsSize = self.itemsArray?.count else { return }
+                    self.itemsArray?.add(newItems)
+                    let arrayWithIndexPaths = NSMutableArray()
+                    for i in 0...(resultsSize + newItems.count) {
+                        arrayWithIndexPaths.add(NSIndexPath(row: i, section: 0))
+                    }
+                    self.collectionView.insertItems(at: arrayWithIndexPaths as! [IndexPath])
+                }, completion: nil)
+            }
+        }, onFailure: { (error) in
+            
+        })
     }
 }
 
-extension NativeViewController: UITableViewDataSource {
-    func numberOfSections(in tableView: UITableView) -> Int {
+extension NativeViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
         return Constants.totalSectionsNative
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return Constants.totalRowsNative
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        guard let itemsArray = itemsArray else { return 0 }
+        return itemsArray.count
     }
     
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        let cell = tableView.dequeueReusableCell(withIdentifier: "TextCell") ?? UITableViewCell(style: .default, reuseIdentifier: "TextCell")
-        cell.textLabel?.text = Constants.nativeText
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "NativeCollectionViewCell", for: indexPath)
+        if let cell = cell as? NativeViewControllerCell {
+            cell.attributionButton.addTarget(self, action: #selector(clickedTaboolaAttribution), for: .touchDown)
+            guard let itemsArray = itemsArray else { return cell }
```

Comment:
again, if you decide it's worth clarifying in code that the left itemsArray is different than the right, it should apply here as well.